### PR TITLE
Icon: Update icon sizing

### PR DESCRIPTION
**Pull request by jliotta**
Per a request from Joey and AJ, we're updating `Icon` sizing to be more like `Typography`. Will run a codemod in Cambly-FE after this is merged to update the sizing of our syntax icons

### CODE REVIEW

File: packages/syntax-core/src/Chip/Chip.tsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ const Chip = forwardRef<HTMLButtonElement, ChipProps>(
         aria-pressed={selected}
         onClick={onChange}
       >
-        {Icon && <Icon className={iconStyles} size="sm" />}
+        {Icon && <Icon className={iconStyles} size={100} />}
         <Box paddingX={Icon ? 1 : 0}>
```
Comment (jliotta) [inline packages/syntax-core/src/Chip/Chip.tsx:127]: @christianvuerings @somethiiing The issue I'm getting with these other components that use `Icon` like `Chip`, `IconButton` etc is that we are still using both Material icons and our own Icon.

Now with this change, the sizing of them is different. Material uses `sm` `med` `lg` and our Icon component will now use `100` `200` etc. 

How do I determine what type of Icon is being passed in as a prop?

Comment (somethiiing) [reply packages/syntax-core/src/Chip/Chip.tsx:127]: cant you just pass in both props?

Comment (jliotta) [reply packages/syntax-core/src/Chip/Chip.tsx:127]: They both use the name `size` as the prop.

I think I got it to work though. The `size` prop that's passed down controls the size for our internal icons and for material the className that was being used controls the size of the icon

File: packages/syntax-core/src/Icon/Icon.tsx [hunk 2]
```diff
@@ -14,14 +14,20 @@ type IconProps = {
   color?: ComponentProps<typeof Typography>["color"];
   /**
    * The size of the Icon.
-   * * `xs`: 16px x 16px
-   * * `sm`: 20px x 20px
-   * * `md`: 24px x 24px
-   * * `lg`: 32px x 32px
+   * * 100: 16px x 16px
```
Comment (somethiiing) [inline packages/syntax-core/src/Icon/Icon.tsx:17]: missing the size 0?

**Review by somethiiing — APPROVED** [review]

File: packages/syntax-core/src/Icon/Icon.tsx (hunk 2 above)
Comment (jliotta) [reply packages/syntax-core/src/Icon/Icon.tsx:17]: Joey didn't spec out a size 0. we can easily add it in the future if that changes

File: packages/syntax-core/src/IconButton/IconButton.tsx [hunk 3]
```diff
@@ -8,12 +8,18 @@ import { backgroundColor } from "../colors/backgroundColor";
...
 import type InternalIcon from "../Icon/Icon";
 
-const iconSize = {
+const materialIconSize = {
   sm: styles.smIcon,
   md: styles.mdIcon,
   lg: styles.lgIcon,
 };
 
+const internalIconSize = {
```
[the remark this answers is not in the document]
Comment (jliotta) [reply packages/syntax-core/src/IconButton/IconButton.tsx]: yeah thanks for the catch

**Pull request merged**